core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -536,8 +537,36 @@ void checkTaskTimeout(TaskDef taskDef, Task task) {
             return;
         }
 
-        String reason = String.format("Task timed out after %d seconds. Timeout configured as %d. "
-            + "Timeout policy configured to %s", elapsedTime/1000L, timeout, taskDef.getTimeoutPolicy().name());
+        String reason = String.format("Task timed out after %d seconds. Timeout configured as %d seconds. "
+            + "Timeout policy configured to %s", elapsedTime / 1000L, timeout / 1000L, taskDef.getTimeoutPolicy().name());
+        timeoutTaskWithTimeoutPolicy(reason, taskDef, task);
+    }
+
+    @VisibleForTesting
+    void checkTaskPollTimeout(TaskDef taskDef, Task task) {
+        if (taskDef == null) {
+            LOGGER.warn("Missing task definition for task:{}/{} in workflow:{}", task.getTaskId(), task.getTaskDefName(), task.getWorkflowInstanceId());
+            return;
+        }
+        if (taskDef.getPollTimeoutSeconds() == null || taskDef.getPollTimeoutSeconds() <= 0 || !task.getStatus().equals(SCHEDULED)) {
+            return;
+        }
+
+        final long pollTimeout = 1000L * taskDef.getPollTimeoutSeconds();
+        final long now = System.currentTimeMillis();
+        final long pollElapsedTime = now - (task.getScheduledTime() + ((long) task.getStartDelayInSeconds() * 1000L));
```

Comment:
When a task is sent back from the worker with `callbackAfterSeconds` set and `IN_PROGRESS` status, Conductor will mark this task as SCHEDULED and set it to invisible state in the queue to be available for polling after the specified callbackAfterSeconds. [Reference](https://github.com/Netflix/conductor/blob/master/core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java#L816-L819)

In these cases, the `pollElapsedTime` calculation here would wrongly mark this task to be poll timed out. This calculation would need to take into account, this scenario of callbackAfterSeconds and invisibility period on the queues. A similar time calculation can be found in the `isResponseTimedOut` method in this class for reference.

Reply:
Thanks for the feedback. I have added the callbackAfterSeconds to the timeout for a correct computation as per your suggestion.